quill-jdbc-zio/src/main/scala/io/getquill/context/zio/ZioJdbcContext.scala

```diff
@@ -0,0 +1,296 @@
+  with JdbcRunContext[Dialect, Naming]
+  with StreamingContext[Dialect, Naming]
+  with ZioPrepareContext[Dialect, Naming]
+  with ZioTranslateContext {
+
+  override private[getquill] val logger = ContextLogger(classOf[ZioJdbcContext[_, _]])
+
+  override type PrepareRow = PreparedStatement
+  override type ResultRow = ResultSet
+  override type RunActionResult = Long
+  override type RunActionReturningResult[T] = T
+  override type RunBatchActionResult = List[Long]
+  override type RunBatchActionReturningResult[T] = List[T]
+
+  // Need explicit return-type annotations due to scala/bug#8356. Otherwise macro system will not understand Result[Long]=Task[Long] etc...
+  override def executeAction[T](sql: String, prepare: Prepare = identityPrepare): RIO[BlockingConnection, Long] =
+    super.executeAction(sql, prepare)
+  override def executeQuery[T](sql: String, prepare: Prepare = identityPrepare, extractor: Extractor[T] = identityExtractor): RIO[BlockingConnection, List[T]] =
+    super.executeQuery(sql, prepare, extractor)
+  override def executeQuerySingle[T](sql: String, prepare: Prepare = identityPrepare, extractor: Extractor[T] = identityExtractor): RIO[BlockingConnection, T] =
+    super.executeQuerySingle(sql, prepare, extractor)
+  override def executeActionReturning[O](sql: String, prepare: Prepare = identityPrepare, extractor: Extractor[O], returningBehavior: ReturnAction): RIO[BlockingConnection, O] =
+    super.executeActionReturning(sql, prepare, extractor, returningBehavior)
+  override def executeBatchAction(groups: List[BatchGroup]): RIO[BlockingConnection, List[Long]] =
+    super.executeBatchAction(groups)
+  override def executeBatchActionReturning[T](groups: List[BatchGroupReturning], extractor: Extractor[T]): RIO[BlockingConnection, List[T]] =
+    super.executeBatchActionReturning(groups, extractor)
+  override def prepareQuery[T](sql: String, prepare: Prepare, extractor: Extractor[T] = identityExtractor): RIO[BlockingConnection, PreparedStatement] =
+    super.prepareQuery(sql, prepare, extractor)
+  override def prepareAction(sql: String, prepare: Prepare): RIO[BlockingConnection, PreparedStatement] =
+    super.prepareAction(sql, prepare)
+  override def prepareBatchAction(groups: List[BatchGroup]): RIO[BlockingConnection, List[PreparedStatement]] =
+    super.prepareBatchAction(groups)
+
+  override protected val effect: Runner = Runner.default
+
+  /** ZIO Contexts do not managed DB connections so this is a no-op */
+  override def close(): Unit = ()
+
+  protected def withConnection[T](f: Connection => Result[T]): Result[T] = throw new IllegalArgumentException("Not Used")
+
+  // Primary method used to actually run Quill context commands query, insert, update, delete and others
+  override protected def withConnectionWrapped[T](f: Connection => T): RIO[BlockingConnection, T] =
+    blocking(RIO.fromFunction((conn: BlockingConnection) => f(conn.get)))
```

Comment:
`f(conn.get)` cloud throw exception. Would be better wrapper that with some `effect`